### PR TITLE
Refactor move nav-native -> SDK type mapping and alert models to core module

### PR DESCRIPTION
**Pull request by Guardiola31337**
### Description
Fixes https://github.com/mapbox/mapbox-navigation-android/issues/3735

Refactor moves nav-native -> SDK type mapping and `alert` models to core module

### Changelog

```
<changelog>Refactored moved nav-native -> SDK type mapping and `alert` models to core module.</changelog>
```

### CODE REVIEW

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt [hunk 1]
```diff
@@ -496,19 +502,31 @@ internal class MapboxTripSession(
             if (!isActive) {
                 return@launch
             }
-            updateEnhancedLocation(status.enhancedLocation, status.keyPoints)
+            val enhancedLocation = status.navigationStatus.location.toLocation()
+            val keyPoints = status.navigationStatus.key_points.toLocations()
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:506]: Wondering if we should move these operations to `ioJobController` 🤔 

Same applies to other ones below 👇 

Any opinions @korshaknn @LukasPaczos?

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/RoadObjectsStore.kt [hunk 2]
```diff
@@ -40,7 +45,7 @@ class RoadObjectsStore internal constructor(
      */
     fun getRoadObjectMetadata(roadObjectId: String): EHorizonObjectMetadata? {
         return navigator.roadObjectsStore?.getRoadObjectMetadata(roadObjectId)?.let {
-            navigator.navigatorMapper.run {
+            run {
```
Comment (Guardiola31337) [inline libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/RoadObjectsStore.kt:48]: Do we need this `run` at all? Do you remember why was added in the first place @korshaknn?

**Comment by korshaknn** [comment]
@Guardiola31337 let's merge current PR `after` https://github.com/mapbox/mapbox-navigation-android/pull/4109
because the most part of the current PR is import changes and it's much easier than fixing all conflicts in my PR.
ok?

File: libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt (hunk 1 above)
Comment (LukasPaczos) [reply libnavigation-core/src/main/java/com/mapbox/navigation/core/trip/session/MapboxTripSession.kt:506]: Those are rather light, I don't think it's worth it.

**Comment by Guardiola31337** [comment]
Closing in favor of https://github.com/mapbox/mapbox-navigation-android/pull/4141

**Pull request closed without merging**